test/sql_test.rb

```diff
@@ -196,6 +196,12 @@ def test_includes
     assert Product.search("product", includes: [:store]).first.association(:store).loaded?
   end
 
+  def test_references
+    skip unless defined?(ActiveRecord)
+    store_names ["Product A"]
+    assert Product.search("product", includes: [:store], references: [:store]).first.association(:store).loaded?
+  end
```

Comment:
Wasn't sure how to test that specifying `references` produces the appropriate JOIN clause, but did want to demonstrate that specification of `references` does not introduce regressions with respect to `includes`.